src/style/style_layer/raster-data-driven_style_layer.js

```diff
@@ -0,0 +1,28 @@
+// @flow
+
+import StyleLayer from '../style_layer';
+
+import properties from './raster-data-driven_style_layer_properties';
+import { Transitionable, Transitioning, PossiblyEvaluated } from '../properties';
+
+import type {PaintProps} from './raster-data-driven_style_layer_properties';
+import type {LayerSpecification} from '../../style-spec/types';
+
+class RasterDataDrivenStyleLayer extends StyleLayer {
+     _lookupTexture: ImageData;
+
+    _transitionablePaint: Transitionable<PaintProps>;
+    _transitioningPaint: Transitioning<PaintProps>;
+    paint: PossiblyEvaluated<PaintProps>;
+
+    constructor(layer: LayerSpecification) {
+        super(layer, properties);
+        this._lookupTexture = layer.lookupTexture;
```

Comment:
If the budget permits it, we should figure out a way to make this an actual paint-property, this would simplify a lot.

Reply:
I think this can be modelled like `heatmap-color` in the `heatmap` layer: https://www.mapbox.com/mapbox-gl-js/style-spec#paint-heatmap-heatmap-color. Basically the user will provide a look-up table/function from which we internally create the look-up texture, instead of generating it on the outside. The implementation of `heatmap` color does the same IRC.

Reply:
But instead of mapping from density values to colors, we map from pixel values to colors.